src/core/meshcop/joiner.hpp

```diff
@@ -109,12 +109,28 @@ class Joiner : public InstanceLocator
      */
     void GetJoinerId(Mac::ExtAddress &aJoinerId) const;
 
+    /**
+     * This method validates the PSKd.
+     *
+     * Per Thread specification, a Joining Device Credential is encoded as
+     * uppercase alphanumeric characters (base32-thread: 0-9,A-Y excluding
+     * I,O,Q, and Z for readability) with a minimum length of 6 such
+     * characters and a maximum length of 32 such characters.
+     *
+     * @retval OT_ERROR_NONE          The PSKd is valid.
+     * @retval OT_ERROR_INVALID_ARGS  The PSKd is invalid.
+     *
+     */
+    static otError ValidatePskd(const char *aPskd);
```

Comment:
nit: how about:
```suggestion
    static bool IsPskdValid(const char *aPskd);
```

Reply:
I have considered this myself. But, if we can implement https://github.com/openthread/openthread/issues/4734, returning `otError` can give detials about why the `PSKd` is considered invalid (e.g. whether it exceed the length range or it includes invalid characters).

Reply:
updated to return`bool`. We can consider returning `otError` when we are going to implement https://github.com/openthread/openthread/issues/4734. @jwhui please review.